src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -1338,6 +1339,29 @@ public async Task ExceptionControllingErrorCode()
         Assert.Equal(2, exception.ErrorCode);
     }
 
+    [Fact]
+    public async Task FormatterNonFatalException()
```

Comment:
We need another test like this that makes the exception happen on the *server* side, and we should assert that that *kills* the connection since a failure to send a response is fatal to JSON-RPC.